spirv_msl.cpp

```diff
@@ -4587,6 +4594,31 @@ void CompilerMSL::bitcast_to_builtin_store(uint32_t, std::string &, const SPIRTy
 {
 }
 
+std::string CompilerMSL::to_expression(uint32_t id)
```

Comment:
This feels a bit ugly, considering all the workarounds needed to pick either GLSL or MSL versions of to_expression above.

In the GLSL backend, there is bitcast_from_builtin_load and bitcast_to_builtin_store which I think would isolate this problem better. These are read-only variables, and they must be loaded first to be used, so we should be able to bitcast in OpLoad if there's a mismatch between builtin and declared type instead of hacking to_expression.